### PR TITLE
Feature/no reference/add error boundary

### PR DESCRIPTION
**Pull request by TomaVadim**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,8 +1,16 @@
 import { Provider } from "react-redux";
 import { PropsWithChildren } from "react";
 
+import { ErrorBoundary } from "react-error-boundary";
```
Comment (yurasytnyk) [inline src/App.tsx:3]: move import from 4 line to 3

File: src/App.tsx [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+
+import { Fallback } from "./features/components/fallback/fallback";
```
Comment (yurasytnyk) [inline src/App.tsx]: rename to ErrorFallback and move to src/components, use import alias @/components/error-fallback

File: src/features/components/fallback/fallback.module.css [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+.button {
+  margin-top: 3rem;
+  min-height: 40px;
+  padding-inline: 30px;
+  border-radius: 5px;
+  background-color: #4c7e35;
+  color: #fff;
+}
```
Comment (yurasytnyk) [inline src/features/components/fallback/fallback.module.css]: add 26 line

File: src/features/components/fallback/fallback.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { FC } from "react";
+
+import style from "./fallback.module.css";
```
Comment (yurasytnyk) [inline src/features/components/fallback/fallback.tsx]: @/components/error-fallback/error-fallback.module.css

File: src/features/components/fallback/fallback.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+type Props = {
```
Comment (yurasytnyk) [inline src/features/components/fallback/fallback.tsx]: interface Props {
}

File: src/features/components/fallback/fallback.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  error: Error;
+  resetErrorBoundary: () => void;
+};
+
+export const Fallback: FC<Props> = ({ error, resetErrorBoundary }) => {
+  return (
+    <div className={style.container} role="alert">
+      <h2>Something went wrong:</h2>
```
Comment (yurasytnyk) [inline src/features/components/fallback/fallback.tsx]: h2
Enter
pre
Enter
button

Comment (yurasytnyk) [reply src/features/components/fallback/fallback.tsx]: add title and reset button text props

Comment (yurasytnyk) [reply src/features/components/fallback/fallback.tsx]: replace div by section

File: src/features/components/fallback/fallback.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { FC } from "react";
+
+import style from "./fallback.module.css";
+
+type Props = {
+  error: Error;
+  resetErrorBoundary: () => void;
```
Comment (yurasytnyk) [inline src/features/components/fallback/fallback.tsx]: let's use more generic name: onErrorReset

File: src/features/components/fallback/fallback.module.css (hunk 3 above)
Comment (TomaVadim) [reply src/features/components/fallback/fallback.module.css]: resolved

File: src/features/components/fallback/fallback.tsx (hunk 5 above)
Comment (TomaVadim) [reply src/features/components/fallback/fallback.tsx]: resolved

File: src/features/components/fallback/fallback.tsx (hunk 7 above)
Comment (TomaVadim) [reply src/features/components/fallback/fallback.tsx]: resolved

File: src/features/components/fallback/fallback.tsx (hunk 6 above)
Comment (TomaVadim) [reply src/features/components/fallback/fallback.tsx]: resolved

File: src/App.tsx (hunk 2 above)
Comment (TomaVadim) [reply src/App.tsx]: resolved

File: src/features/components/fallback/fallback.tsx (hunk 4 above)
Comment (TomaVadim) [reply src/features/components/fallback/fallback.tsx]: resolved

File: src/App.tsx (hunk 1 above)
Comment (TomaVadim) [reply src/App.tsx:3]: resolved

File: src/components/error-fallback/error-fallback.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+interface Props {
+  error: Error;
+  onErrorReset?: () => void;
+};
+
+export const ErrorFallback: FC<Props> = ({ error, onErrorReset }) => {
+  return (
+    <section role="alert">
+      <h2>Something went wrong:</h2>
```
Comment (yurasytnyk) [inline src/components/error-fallback/error-fallback.tsx:11]: add title prop

File: src/components/error-fallback/error-fallback.tsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+      <pre>{error.message}</pre>
+
+      <button onClick={onErrorReset}></button>
```
Comment (yurasytnyk) [inline src/components/error-fallback/error-fallback.tsx:15]: add resetButtonText prop, but the better approach would be if you move the button out of the component and pass it as children

File: src/components/error-fallback/error-fallback.tsx [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import { FC } from "react";
+
+interface Props {
+  error: Error;
+  onErrorReset?: () => void;
+};
```
Comment (yurasytnyk) [inline src/components/error-fallback/error-fallback.tsx:6]: remove ;

Comment (yurasytnyk) [reply src/components/error-fallback/error-fallback.tsx:6]: resolved

File: src/components/error-fallback/error-fallback.tsx (hunk 9 above)
Comment (yurasytnyk) [reply src/components/error-fallback/error-fallback.tsx:15]: resolved

File: src/components/error-fallback/error-fallback.tsx (hunk 8 above)
Comment (yurasytnyk) [reply src/components/error-fallback/error-fallback.tsx:11]: resolved

**Pull request closed without merging**